### PR TITLE
#221 | Pagination Infrastructure Has Been Refactored and Pagination Validations Have Been Fixed

### PR DESCRIPTION
**Pull request by agitrubard**
### Description

- Pagination Infrastructure Has Been Refactored and Pagination Validations Have Been Fixed
- `@Valid` Annotation Has Been Added to Parameter of `AdminUserRegisterApplicationController.getRegistrationApplications()` Method
- `@UUID` Annotation Has Been Added to Parameter of `AdminUserRegisterApplicationController.getRegistrationApplicationById()` Method

### Checklist

Please check the following before submitting your pull request:

- [X] I have tested my changes locally and they are working as expected.
- [X] The code is formatted according to the project's coding guidelines and style.
- [X] Necessary documentation has been added or existing documentation has been updated.
- [X] Relevant unit tests have been written and included.
- [X] Relevant integration tests have been written and included.
- [X] The code has been reviewed to ensure its quality.
- [X] The code does not contain any issues flagged by SonarLint.
- [X] Default reviewers have been assigned to this pull request.
- [X] Assignees have been added if necessary.
- [X] Labels have been applied if necessary.
- [X] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

### Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

**Review by anilhazar — APPROVED** [review]

**Review by AhmetAksunger — DISMISSED** [review]

File: src/main/java/com/ays/common/model/AysPaging.java [hunk 1]
```diff
@@ -1,58 +1,39 @@
...
 import lombok.Setter;
 import org.hibernate.validator.constraints.Range;
 
 /**
- * A class representing paging parameters for a query. It includes the page number and page size.
- * The page number is 1-based, but internally, it is stored as 0-based to comply with Spring Data's pagination system.
+ * The class `Paging` is a data model used for pagination operations. It includes essential
+ * pagination parameters such as page number and page size.
+ * <p>
+ * The page number (`page`) should be a value between 1 and 99999999.
```
Comment (skayikci) [inline src/main/java/com/ays/common/model/AysPaging.java:11]: field üzerinde zaten tanımlama yapilmis. tekrar buraya comment olarak eklemeye gerek var mi?

Comment (agitrubard) [reply src/main/java/com/ays/common/model/AysPaging.java]: Eklemeyebiliriz evet, kaldırdım 👍🏼

**Comment by agitrubard** [comment]
Build aşamasını bozacak bir değişiklik yapmadım ama müsait olunca buraya bakacağım, bilginize, o zamana kadar draft'a alıyorum

**Comment by agitrubard** [comment]
Bu branch'te hata aldığım için, pull requesti kapatıyorum. Yeni bir branch üzerinden pull request açacağım.

**Pull request closed without merging**